src/IO/H5/Helpers.cpp

```diff
@@ -473,8 +473,15 @@ void write_to_attribute(const hid_t location_id, const std::string& name,
   const hid_t att_id = H5Acreate2(location_id, name.c_str(), h5_type<Type>(),
                                   space_id, h5p_default(), h5p_default());
   CHECK_H5(att_id, "Failed to create attribute '" << name << "'");
-  CHECK_H5(H5Awrite(att_id, h5_type<Type>(), static_cast<const void*>(&value)),
-           "Failed to write value: " << value);
+  if constexpr (std::is_same_v<Type, std::string>) {
+    const auto cstr = value.c_str();
+    CHECK_H5(H5Awrite(att_id, h5_type<Type>(), &cstr),
```

Comment:
From the `h5_type` docs: "\note For strings, the returned type must be released with H5Tclose"

Reply:
This seems like it isn't always done in the `Helpers.cpp`

Reply:
Yes, `read_value_attribute` also gets it wrong.  The other calls aren't strings.

Reply:
Will is correct, this is a latent bug in the other code as well. It only matters for strings since we set some properties on the string to make it nicer to deal with.

Reply:
Ok so shall I change it here, or do a follow-up?

Reply:
Fix it in this PR.

Reply:
Squashed it in to the "Write H5" commit since it was only a few lines